File: Programmers/Lv. 2/조이스틱.py
def solution(name):
    alpha = ["A","B","C","D","E","F","G","H","I","J","K","L","M","N","O","P","Q","R","S","T","U","V","W","X","Y","Z"]

    answer = 0

    for i in range(len(name)):
        temp = alpha.index(name[i])           
        answer += min(temp, 26 - temp)   

    pos = 1
    temp = []
    while pos < len(name):
        if name[pos] == "A":
            s = pos
            c = 0
            while pos < len(name):
                if name[pos] == "A":
                    pos += 1
                    c += 1
                else:
                    break
            temp.append([s, c])
        pos += 1

    m = len(name) - 1
    for t in temp:
        a = (t[0]-1)*2 + len(name) - (t[0] + t[1])
        b = (len(name) - (t[0] + t[1]))*2 + t[0]-1

        m = min([m, a, b])

    answer += m

    return answer

File: Programmers/Lv. 2/조이스틱.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

map<char, int> alphabet = {
    {'A', 0}, {'B', 1}, {'C', 2}, {'D', 3}, {'E', 4}, {'F', 5}, {'G', 6}, {'H', 7},
    {'I', 8}, {'J', 9}, {'K', 10}, {'L', 11}, {'M', 12}, {'N', 13}, {'O', 12}, {'P', 11},
    {'Q', 10}, {'R', 9}, {'S', 8}, {'T', 7}, {'U', 6}, {'V', 5}, {'W', 4}, {'X', 3},
    {'Y', 2}, {'Z', 1}
};

int solution(string name) {
    int answer = alphabet[name[0]];
    
    vector<pair<int, int>> v;
    int pos = 0;
    int c = 0;
    for (int i = 1; i < name.length(); i++)
    {
        answer += alphabet[name[i]];
        
        if (name[i] == 'A')
        {            
            if (c == 0)
            {
                pos = i;
            }
            c++;
            
            if (i == name.length() - 1)
            {
                v.push_back({pos, c});
            }
        }
        else
        {
            if (c != 0)
            {
                v.push_back({pos, c});
                c = 0;
            }
        }
    }

    int l = name.length();
    
    vector<int> mins = {l-1};
    
    for (auto pair : v)
    {
        
        int a = pair.first;
        int b = pair.second;
        
        int min1 = (a-1)*2 + l - (a + b);
        int min2 = (l-(a+b))*2 + a - 1;
        
        mins.push_back(min1);
        mins.push_back(min2);
    }
    
    int m = *min_element(mins.begin(), mins.end());
    
    return answer + m;
}